vyper/cli/vyper_compile.py

```
@@ -135,16 +138,33 @@ def _parse_args(argv):
         args.ovm,
     )
 
-    if output_formats == ("combined_json",):
-        print(json.dumps(compiled))
-        return
 
-    for contract_data in compiled.values():
-        for data in contract_data.values():
-            if isinstance(data, (list, dict)):
-                print(json.dumps(data))
-            else:
-                print(data)
+    if not args.output_path:
+        if output_formats == ("combined_json",):
+            print(json.dumps(compiled))
+            return
+
+        for contract_data in compiled.values():
+            for data in contract_data.values():
+                if isinstance(data, (list, dict)):
+                    print(json.dumps(data))
+
+    else:
+        if output_formats == ("combined_json",):
+            with open(args.output_path,"w") as f:
+                f.write(json.dumps(compiled))
+                return
+                
+        for contract_data in compiled.values():
+            for data in contract_data.values():
+                if isinstance(data, (list, dict)):                    
+                    with open(args.output_path,"w") as f:
+                        f.write(json.dumps(data))
+                elif "bytecode" in contract_data.keys():
+                    with open(args.output_path,"w") as f:
+                        f.write(contract_data["bytecode"])
+                else:
+                    print("unrecognized output")
```

Please throw an exception here instead